server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -3183,7 +3184,7 @@ public DiskOffering updateDiskOffering(final UpdateDiskOfferingCmd cmd) {
             throw new InvalidParameterValueException(String.format("Unable to update disk offering: %s by id user: %s because it is not root-admin or domain-admin", diskOfferingHandle.getUuid(), user.getUuid()));
         }
 
-        final boolean updateNeeded = name != null || displayText != null || sortKey != null || displayDiskOffering != null;
+        final boolean updateNeeded = name != null || displayText != null || sortKey != null || displayDiskOffering != null || tags != null;
```

Comment:
Can you extract this `name != null || displayText != null || sortKey != null || displayDiskOffering != null || tags != null` to a method? Then, you can document it (explain what it checks, and why), and after that, you can write unit tests for all of the cases that the method must return true/false.

Moreover, instead of checking if it is different from null, you could use `String.Utils.isNotBlank`

Reply:
I extract this to an method, and add documentation for this new method